graphdatascience/query_runner/neo4j_query_runner.py

```diff
@@ -108,6 +115,22 @@ def run_query_with_logging(
             else:
                 return future.result()
 
+    def _forward_cypher_warnings(self, notification: Dict[str, Any]) -> None:
+        # (see https://neo4j.com/docs/status-codes/current/notifications/ for more details)
+        severity = notification["severity"]
+        if severity == "WARNING":
+            # FIXME handle the field deprecations
+            if "query used a deprecated field from a procedure" in notification["description"]:
+                return
```

Comment:
The problem is that uses cannot specify the yields upfront.
We could decide for the user to not return the deprecated fields by default, and they can set a flag to get them, but this could break code.

Alternatively we could have the warning by default and allow the user to trim down further

Reply:
By definition in python. `warnings.warn` is for warnings users can solve, whereas `log.warn` would be for warnings people cant fix.

IMO, we could then just log to let the user know.

Reply:
Sounds good to me

Reply:
The main downside is that the log becomes a bit noisy for the user.
Split between letting users know that we could handle this better or we first built the YIELD feature and then actually use warnings

Reply:
Let's break the tie in favour of the conservative option and not make a change.